## 1. Introduction

modification test

### Table of contents

- 1. Introduction

- 2. Exploratory data analysis
- 2. 1 Importing Libraries and load the data
- 3. Preprocessing
- 3. 1 Cleaning the data 
- 3. 2 Handling missing values
- 3. 3 Converting text labels to numbers (feature encoding)
- 4. Modelling
- 4. 1 symple base-line model
- 4. 2 two sophisticated model approaches
- 4. 3 Experiment and testing the model
- 5. Results
- 6. Discussion


## 2. Exploratory data analysis

Analyse your data. Visualise and explain the data features you deem to be relevant for
the project.

### 2.1 Importing Libraries and load data

In [11]:
# Importing the libraries
import pandas as pd
import matplotlib.pyplot as plt
from sklearn import preprocessing
import numpy as np

heart_data = pd.read_csv("Heart_Disease_Prediction.csv")

heart_data.head()

index  Age  Sex  Chest pain type   BP  Cholesterol  FBS over 120  \
0      0   70    1                4  130          322             0   
1      1   67    0                3  115          564             0   
2      2   57    1                2  124          261             0   
3      3   64    1                4  128          263             0   
4      4   74    0                2  120          269             0   

   EKG results  Max HR  Exercise angina  ST depression  Slope of ST  \
0            2     109                0            2.4            2   
1            2     160                0            1.6            2   
2            0     141                0            0.3            1   
3            0     105                1            0.2            2   
4            2     121                1            0.2            1   

   Number of vessels fluro  Thallium Heart Disease  
0                        3         3      Presence  
1                        0         7       Absence  
2                        0         7      Presence  
3                        1         7       Absence  
4                        1         3       Absence

## 3. Preprocessing

Explain what kind of preprocessing, feature encoding you are applying

## 3.1 Cleaning the data

In [ ]:
#drop the 'id' column
heart_data = heart_data.drop(columns=['id'])



## 3.3 Converting text labels to numbers (feature encoding)

In [ ]:
##Target encoding for absence and presence of heart disease

## Feature encoding for binary categorical features and continuous features

## Encoding

what needs to be encoded:
    Sex binary          category
    Chest pain type     unordered categories
    FBS over 120        binary
    EKG results         categories
    Exercise angina     binary
    Slope of ST         categories
    Thallium            categories

do not one hot enocde:
Age                         continuous
BP
Cholesterol
Max HR
ST depression
Number of vessels fluro     ordered count (0–3)




## 4. Modelling

### 4. 1 symple base-line model

perfect model setup for baseline: Logicstic regression

is interpretable, medically standard and easy to explain coefficients

### 4. 2 two sophisticated model approaches

number 1 random forest: captures feature interactions, non linear splits, robust on small datasets

numeber 2 Gradient Boosting or SVM (RBF)
